### DISCUSSION TITLE
Pass object with v-model or only the index

### DISCUSSION THREAD

**Discussion by Nisgrak** [vuejs/pinia#1419] [Help and Questions]
Hi!

I have the state of the app in pinia similar to this:
```json
{
  "sections": [
    {
      "instruments": [
        {
          "notes": [0, 1, 5, 2]
        }
      ]
    }
  ]
}
```

Each section is a component, that have 1 children for each instrument, that have 1 children for each note (it's a musical app)

Actually I'm passing the model-value and using the @update event with a complex function to adapt the object, would be a better approach to pass the indexSection, indexInstrument and indexNote and read/write directly to the store by each children?

The v-model way works, but looks a bit messy.

What do you think?

**Comment by posva** [comment]
Using `v-model` would be more idiomatic and flexible (from a component perspective):

```vue
<MyNote v-model:note="store.sections[sectionIndex].instrument[instrumentIndex].notes[noteIndex]" />
```

Depending on the other things you are doing, you might have a getter that returns the currently selected section / instrument, so the code above becomes


```vue
<MyNote v-for="(note, i) in currentInstrument.notes v-model:note="currentInstrument.notes[i]" />
```
